webapp/channels/src/components/move_thread_modal/move_thread_modal.tsx

```diff
@@ -0,0 +1,256 @@
+
+export type ActionProps = {
+
+    // join the selected channel when necessary
+    joinChannelById: (channelId: string) => Promise<ActionResult>;
+
+    // switch to the selected channel
+    switchToChannel: (channel: Channel) => Promise<ActionResult>;
+
+    // action called to move the post from the original channel to a new channel
+    moveThread: (postId: string, channelId: string) => Promise<ActionResult>;
+}
+
+export type OwnProps = {
+
+    // The function called immediately after the modal is hidden
+    onExited?: () => void;
+
+    // the post that is going to be moved
+    post: Post;
+};
+
+export type Props = OwnProps & {actions: ActionProps};
+
+const noop = () => {};
+
+const getChannel = makeGetChannel();
+
+const MoveThreadModal = ({onExited, post, actions}: Props) => {
+    const {formatMessage} = useIntl();
+
+    const originalChannel = useSelector((state: GlobalState) => getChannel(state, {id: post.channel_id}));
+    const currentTeam = useSelector(getCurrentTeam);
+
+    const [bodyHeight, setBodyHeight] = useState<number>(0);
+    const [hasError, setHasError] = useState<boolean>(false);
+    const [postError, setPostError] = useState<React.ReactNode>(null);
+    const [selectedChannel, setSelectedChannel] = useState<ChannelOption>();
+    const [isButtonClicked, setIsButtonClicked] = useState<boolean>(false);
+
+    const bodyRef = useRef<HTMLDivElement>();
+
+    const measuredRef = useCallback((node) => {
+        if (node !== null) {
+            bodyRef.current = node;
+            setBodyHeight(node.getBoundingClientRect().height);
+        }
+    }, []);
+
+    const onHide = useCallback(() => {
+        onExited?.();
+    }, [onExited]);
+
+    const handleChannelSelect = useCallback(
+        (channel: ValueType<ChannelOption>) => {
+            if (Array.isArray(channel)) {
+                setSelectedChannel(channel[0]);
+            }
+            setSelectedChannel(channel as ChannelOption);
+        },
+        [],
+    );
+
+    // since the original post has a click handler specified we should prevent any action here
+    const preventActionOnPreview = (e: React.MouseEvent) => {
+        e.preventDefault();
+        e.stopPropagation();
+    };
+
+    const messagePreviewTitle = formatMessage({
+        id: 'move_thread_modal.preview.title',
+        defaultMessage: 'Message preview',
+    });
+
+    const previewMetaData: PostPreviewMetadata = {
+        post,
+        post_id: post.id,
+        team_name: currentTeam.name,
+        channel_display_name: originalChannel.display_name,
+        channel_type: originalChannel.type,
+        channel_id: originalChannel.id,
+    };
+
+    const notificationText = (
+        <FormattedMessage
+            id='move_thread_modal.notification.dm_or_gm'
+            defaultMessage='Moving this thread changes who has access'
+            values={{
+                strong: (x: React.ReactNode) => <strong>{x}</strong>,
+            }}
```

Comment:
I see no strong tag on the default message, so this may not be needed.

Also, curious about the inconsistency between using `formatMessage` a few lines above, but using `FormattedMessage` here. I am biased towards the `formatMessage`  approach, but my main question is why not be consistent.

Reply:
Removed. The team pulled almost all of the move thread modal logic from the forward post modal. Most of what you're seeing here is from that modal, including the inconsistency between formatMessage and FormattedMessage.

The notificationText specifically is done with FormattedMessage to be a JSX element for the NotificationBox component